core/src/main/java/com/airbnb/aerosolve/core/features/SparseLabeledPoint.java

```diff
@@ -0,0 +1,19 @@
+package com.airbnb.aerosolve.core.features;
+
+import lombok.Data;
+
+import java.io.Serializable;
+
+/**
+ * This is a compressed representation of a feature vector indexed via a feature index map. It
+ * captures values in flat representation for space and runtime efficiency during model training.
+ */
+@Data
+public class SparseLabeledPoint implements Serializable {
```

Comment:
Maybe a safe check during construction that indices/values are having the same length?